src/main/java/dk/kb/util/yaml/YAML.java

```diff
@@ -821,59 +841,68 @@ private void traverseList(List<String> yPath, Object yaml, MultipleValuesVisitor
      * The implementation supports conditional lookups as well. This means that a query for zoo.[foo=bar].baz returns
      * all values for baz, that are a child of zoo and has the sibling-key foo with the value bar.
      * @param yPath a list of path elements. This list contains all parts of a specified path, which is most likely
-     *              delivered through the {@link #visit(Object path, YAML yaml)}-method.
+     *              delivered through the {@link #visit(Object path, YAML yaml, YAMLVisitor visitor)}-method.
      * @param yaml the current place in the YAML file being traversed. This should be an instance of a Map.
      * @param visitor used to collect values that match the given path. The only function of this visitor is to collect
      *               values to its internal list of extracted values: {@link MultipleValuesVisitor#extractedValues}.
      */
-    private void traverseMap(List<String> yPath, Object yaml, MultipleValuesVisitor visitor) {
+    private void traverseMap(YPath yPath, Object yaml, MultipleValuesVisitor visitor) {
```

Comment:
No need for this to be the specialized `MultipleValuesVisitor`. Also: This forces the cast in line 772, which makes the implementation less generic.

Reply:
Please also update the Javadoc for the `visitor`-parameter. As it is now general, there's no telling what the purpose of the visitor is.